### PR TITLE
Added position for productTabs, refactored productTabs, added global state for active product tab

### PR DESCRIPTION
**Pull request by neeeeecka**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/5346

**Problem:**
* Previously, it was not possible to (Easily and properly) insert tabs between specific locations as well as it wasn't possible to change active tab's state from anywhere(Usual workaround was to add refs and call click events on tabs)

**In this PR:**
* Created global store `ProductTabsReducer` with `activeTab` property.
* Added `position` property to product tabs indicating their sorting order (Starting with 100, 200, 300, its possible to insert anywhere between these numbers).
* Refactored `ProductTabs` to move all logic to the container.
* Refactored `ProductTab` to use product `id` instead of `name` for handling click events.

### CODE REVIEW

**Comment by neeeeecka** [comment]
Example for inserting tab at any place:
image
Result:
image
Example for setting global state:
```
export const mapDispatchToProps = (dispatch: Dispatch): ProductTabsContainerMapDispatchProps => ({
    setActiveTab: (newActiveTab) => dispatch(updateActiveTab(newActiveTab)),
});
...
Somewhere:
setActiveTab('TEST_TAB');
...
```

File: packages/scandipwa/src/component/ProductTab/ProductTab.component.tsx [hunk 1]
```diff
@@ -12,33 +12,23 @@
...
+    onClick = () => {
+        const { onClick, tab: { id } } = this.props;
 
-    onClick(): void {
-        const { onClick, tabName } = this.props;
-
-        onClick(tabName);
-    }
+        onClick(id);
+    };
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductTab/ProductTab.component.tsx:26]: How are you gonna be a plugin to it? 
As I know arrow functions not working for plugins

File: packages/scandipwa/src/component/ProductTabs/ProductTabs.container.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        if (!tabs?.length) {
+            return;
+        }
+
+        const [{ id }] = tabs;
+
+        this.handleTabSelect(id);
+    }
+
+    render() {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductTabs/ProductTabs.container.tsx:90]: Return type please everywhere

File: packages/scandipwa/src/component/ProductTabs/ProductTabs.container.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        this.updateDefaultSelectedTab();
+    }
+
+    componentDidUpdate(prevProps: ProductTabsContainerProps): void {
+        const { tabs: prevTabs } = prevProps;
+        const { tabs } = this.props;
+        const [prevTab] = prevTabs;
+        const [tab] = tabs;
+
+        if (tab?.id !== prevTab?.id) {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/ProductTabs/ProductTabs.container.tsx:55]: Destruct

File: packages/scandipwa/src/store/ProductTabs/ProductTabs.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    switch (type) {
+    case ProductTabsActionType.UPDATE_ACTIVE_PRODUCT_TAB:
+        return { ...state, activeTab };
+    default:
+        return state;
+    }
+};
+
+export default ProductTabsReducer;
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/ProductTabs/ProductTabs.reducer.ts:41]: Do we really need separate reducer for product tabs? Maybe we can move it to product store?

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Comment by AleksandrsKondratjevs** [comment]
Closing due to deprecated code and no activity.

**Pull request closed without merging**